Zend/zend_language_scanner.l

```diff
@@ -1545,12 +1615,37 @@ NEWLINE ("\r"|"\n"|"\r\n")
 	return T_SR;
 }
 
+<ST_IN_SCRIPTING>","/{OPTIONAL_WHITESPACE}{LABEL}{OPTIONAL_WHITESPACE}"=" {
+	if((in_class && in_const_list)) {
+		yy_push_state(ST_LOOKING_FOR_SR_NAME);
+	}
+	return ',';
+}
+
+<ST_IN_SCRIPTING>";" {
+	if(in_const_list) {
+		in_const_list = 0;
+	}
+	if(in_use_list) {
+		if(!in_class || (in_class && !in_trait_conflict)){
+			in_use_list = 0;
+		}
+	}
+	return ';';
+}
+
 <ST_IN_SCRIPTING>{TOKENS} {
 	return yytext[0];
 }
 
 
 <ST_IN_SCRIPTING>"{" {
+	if(in_class) {
+		if(in_use_list) {
+			in_trait_conflict = 1;
+		}
+		in_class++;
+	}
 	yy_push_state(ST_IN_SCRIPTING);
 	return '{';
 }
```

Comment:
Sigh. So this is valid syntax too... surprised there isn't a single test for that on the test suite AFAIK. After https://github.com/php/php-src/pull/1054#discussion_r25630665 I'm not surprised.

Thanks, I